shared-libs/rules-engine/src/pouchdb-provider.js

```diff
@@ -44,13 +44,15 @@ const medicPouchProvider = db => {
 
     stateChangeCallback: docUpdateClosure(db),
 
-    commitTargetDoc: (assign, userDoc, docTag) => {
-      const userContactId = userDoc && userDoc._id;
-      const _id = `target-${docTag}-${userContactId}`;
+    commitTargetDoc: (assign, userContactDoc, userSettingsDoc, docTag) => {
+      const userContactId = userContactDoc && userContactDoc._id;
+      const userSettingsId = userSettingsDoc && userSettingsDoc._id;
+      const _id = `target-${docTag}-${userContactId}-${userSettingsId}`;
       const createNew = () => ({
         _id,
         type: 'target',
-        user: userContactId,
+        user: userSettingsId,
```

Comment:
if the target doc's `user` attriute is not used for replication, is it still needed at all? can we remove it?

Reply:
I opted to keep it just to be easily readable to humans, since it's more of a pain to try to discern the user's id by visually splitting the target document's uuid. It might also be useful for analytics? 
If you're adamant into keeping these docs as compact as possible, I can remove it, no issues :).

Reply:
i'm just trying to reduce the amount of documents being injected into the public interfaces for each class. it helps keep unit tests clear. this is particularly important for the library because my goal is to use this shared-library in the `medic-conf-test-harness`. new dependencies mean more mocking for tasks&targets tests. 

your point about analytics is good one -- so i'm happy to keep that.

but in the interest of reducing dependencies : the `settings` object that gets injected into the library is actually a custom object. would you consider just adding a `userSettingsDocId` attribute instead of injecting the entire document? i'd be happy to make that change within this PR since i'm somewhat responsible for this bug :)

Reply:
i'm just trying to reduce the amount of documents being injected into the public interfaces for each class. it helps keep unit tests clear. this is particularly important for the library because my goal is to use this shared-library in the medic-conf-test-harness. new dependencies mean more mocking for tasks&targets tests.

your point about analytics is good one -- so i'm happy to keep that.

but in the interest of reducing dependencies : the settings object that gets injected into the library is actually a custom object. would you consider just adding a userSettingsDocId attribute instead of injecting the entire document? i'd be happy to make that change within this PR since i'm somewhat responsible for this bug :)

Reply:
I guess we could move everything into a "context" object or similar, and add the hydrated contact AND the user's ID into that. Good suggestion! Thanks!

Reply:
okay thanks :) check out the format of the `settings` object. it might be a logical and existing container for the "user's settings".

Reply:
along the same lines of thinking to help analtyics... it will probably be quite nice to add a `reportingPeriod` or similar with a value of `docTag`. bonus points available :)

Reply:
I've added `reporting_period` and changed the ids to separate things with a `~`!